CHANGELOG.md

```diff
@@ -4,6 +4,17 @@ All notable changes to this project will be documented in this file.
 
 The format is based on Keep a Changelog and this project adheres to Semantic Versioning.
 
+## [UnReleased] - 2022.xx.xx
+
+### Changed
+
+- Minimal optimization with private properties mangling, macro inlining and const enum. see [#24]
+- Private properties are now mangled.
+- `checkWithinAccessParams` function is removed.
+- constants of `HashContainer` are moved to `HashContainerConst` const enum.
+- The iteratorType parameter in the constructor of `ContainerIterator`, `RandomIterator`, `DequeIterator`, `LinkNode`. `Vector`, `TreeIterator`, `OrderedMapIterator` and `OrderedSetIterator` are now changed from `boolean` type to `IteratorType` const enum type.
```

Comment:
No need to enumerate iterator types.

- The iteratorType parameter in the constructor now changed from `boolean` type to `IteratorType` const enum type.